antarest/study/business/output_variables_management.py

```diff
@@ -0,0 +1,174 @@
+
+from antarest.study.business.aggregator_management import (
+    AggregatorManager,
+    MCAllAreasQueryFile,
+    MCAllLinksQueryFile,
+    MCIndAreasQueryFile,
+    MCIndLinksQueryFile,
+    MCRoot,
+    QueryFileType,
+)
+from antarest.study.storage.output_model import OutputVariablesMetadata
+from antarest.study.storage.rawstudy.model.filesystem.matrix.matrix import MatrixFrequency
+
+
+class OutputVariablesManager:
+    """
+    Manage output variables view.
+    """
+
+    @staticmethod
+    def _filter_files_with_same_prefix(
+        parent_folder: Path, file_type_class: type[QueryFileType]
+    ) -> dict[QueryFileType, MatrixFrequency]:
+        """
+        Iterate over all existing files inside the parent_folder.
+        Returns only one file for each query file class.
+        This way we avoid opening files we know will have the same headers
+
+        Example:
+            In the folder, there's 4 files:
+            `values-hourly.txt`, `values-weekly.txt`, `values-monthly.txt` and `details-annual.txt`
+            The method will only return 1 `values` file and the `details` one.
+
+        Args:
+            - parent_folder: parent folder containing several files
+            - file_type_class: query file type class.
+
+        Returns:
+            - A mapping from file type class to a certain existing matrix frequency
+        """
+        all_files = [d.name for d in parent_folder.iterdir()]
+        file_dict = {}
+        for file in all_files:
+            splitted_file_name = file.removesuffix(".txt").split("-")
+            if len(splitted_file_name) == 2:
+                file_type, freq = splitted_file_name
+            else:
+                file_type1, file_type2, freq = splitted_file_name
+                file_type = f"{file_type1}-{file_type2}"
+            file_dict[file_type_class(file_type)] = MatrixFrequency(freq)
+        return file_dict
+
+    @staticmethod
+    def _read_headers_only(
+        file_path: Path, mc_root: MCRoot, freq: MatrixFrequency, file_type: QueryFileType
+    ) -> dict[str, set[str]]:
+        """
+        Returns the headers of a given output file.
+
+        Args:
+            - file_path: path of the output file
+            - mc_root: mc-ind or mc-all
+            - freq: MatrixFrequency of the given file.
+            - file_type: query file type class.
+
+        Returns:
+            - A mapping containing the dataframe headers
+        """
+        body = AggregatorManager.parse_output_file(file_path, freq, 0)
+
+        if "details" in file_type.value:
+            cols_mapping: dict[str, set[str]] = {}
+            for col in body.columns:
+                cols_mapping.setdefault(col[0], set()).add(col[1])
+            return cols_mapping
+
+        new_cols: dict[str, set[str]] = {}
+        for col in body.columns:
+            if mc_root == MCRoot.MC_IND:
+                name_to_consider = col[0] if file_type.value == "values" else " ".join(col)
```

Comment:
Not sure in which case we join here:
the "details" case is already handled above ?